### ISSUE TITLE
Should MacroCall be MacroDef?

### ISSUE THREAD

**Issue by kjeremy** [rust-analyzer/rust-analyzer#2115]
I was looking into plugging mbe into signature help and while I don't know a lot about macros I noticed that 

```rust
/// empty macro
macro_rules! foo { // <--- Why is this MACRO_CALL ?
    () => {}
}

fn f() {
    foo!(<|>); // <--- other MACRO_CALL
}
```

contains two MACRO_CALL nodes once of which I would think of as the definition. Indeed https://doc.rust-lang.org/reference/macros-by-example.html shows the first example as `MacroRulesDefinition`

**Comment by matklad** [comment]
`macro_rules!` is almost a macro. In particular, the grammar in rustc was like

```
macro_call ::= ident! ident? token_tree
```

I think right now macro calls with extra ident are removed and macro_rules is special cased, but for example, macro_rules body is still parsed as a general token_tree and destcructed into `$( lhs => rhs);*` using the macro by example machinary as well.

TL;DR: it's a macro call because, syntactically, it's actually a macro call.

**Comment by kjeremy** [comment]
Thanks
